bench/bench_queue.ml

```diff
@@ -46,8 +46,13 @@ let run_one ~budgetf ?(n_adders = 2) ?(n_takers = 2)
         let n = Util.alloc n_msgs_to_take in
         if n <> 0 then
           let rec loop n =
-            if 0 < n then
-              loop (n - Bool.to_int (Option.is_some (Queue.pop_opt t)))
+            if 0 < n then begin
+              match Queue.pop_opt t with
+              | None ->
+                  Domain.cpu_relax ();
```

Comment:
`Domain.cpu_relax ()` on OCaml 4 translates to calling `Thread.yield ()` via `domain_shims`.  In older versions of OCaml 4 poll points are not automatically added to all loops.  This means that the emulation of domains by `domain_shims` may break down as a loop that never polls cannot be stopped by the runtime and the program gets stuck.  So, this PR contains a lot of additions of `Domain.cpu_relax ()` calls to loops.  Some of those calls might be unnecessary, but most should be relatively harmless.  We can also remove unnecessary calls, if any, later.